### ISSUE TITLE
model / fields circular references

### ISSUE THREAD

**Issue by franzeal** [formly-js/ng2-formly#346]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request
```

**Current behavior**
In testing the changes to address #344 (see #345) the fixes to the repeatSection's manipulation of the FormArray work as expected against beta 14.  With the latest build of formly, neither the old nor the fixed repeatSection implementations work correctly.  I expect this is a symptom of some other problem with the latest build.

[pre-fixed repeatSection with latest build](https://plnkr.co/edit/VIJtfp4eRSjZFPtE1ZlE?p=preview)
[fixed repeatSection with latest build](https://plnkr.co/edit/5MF719epnQQ7ascJzZHP?p=preview)

**Expected behavior**
[fixed repeatSection with beta 14](https://plnkr.co/edit/QbxPJF3VybsWoHLqzrPP?p=preview)

**Comment by mohammedzamakhan** [comment]
Whats not working, it feels to me that it works, may be I didn't understand it

**Comment by franzeal** [comment]
In the plunkers with the latest build, if you can look at the model when trying to use the repeatSection you should be able to see the problem.

I've commented out the model and fields JSON piped to the view because the JSON pipe will throw errors due to JSON.stringify not liking circular references.  

I believe these are just a symptom of the problem.  I didn't quite know what to attribute or identify this issue as.

**Comment by mohammedzamakhan** [comment]
May be #343 and #345 going together caused this issue.

**Comment by franzeal** [comment]
Doesn't appear that it could have been #345 because the repeatSection prior to that fix has the same problem with the latest build:

https://plnkr.co/edit/VIJtfp4eRSjZFPtE1ZlE?p=preview

**Comment by aitboudad** [comment]
fixed by #347:
https://plnkr.co/edit/VNh8ccoyUJ6GSz7d0jKH?p=preview
